game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, false, false, true, false),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.STANDARD,
+            FirstStrikeBattleStep.DEFENDER));
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has no destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is true, then attacker has return fire none")
+  void firstStrikeAttackerReturnFireAttHasDestroyerDefNoDestroyerWW2v2FalseSneakAttackTrue() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, false, false, true, false),
+        MustFightBattle.ReturnFire.NONE,
+        true);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has no destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is true, then defender has return fire all")
+  void firstStrikeDefenderReturnFireAttHasDestroyerDefNoDestroyerWW2v2FalseSneakAttackTrue() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, false, false, true, false),
+        MustFightBattle.ReturnFire.ALL,
+        false);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has no destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is false, then attacker>standard>defender")
+  void firstStrikeOrderAttHasDestroyerDefNoDestroyerWW2v2FalseSneakAttackFalse() {
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, false, false, false, false),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.STANDARD,
+            FirstStrikeBattleStep.DEFENDER));
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has no destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is false, then attacker has return fire none")
+  void firstStrikeAttackerReturnFireAttHasDestroyerDefNoDestroyerWW2v2FalseSneakAttackFalse() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, false, false, false, false),
+        MustFightBattle.ReturnFire.NONE,
+        true);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has no destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is false, then defender has return fire all")
+  void firstStrikeDefenderReturnFireAttHasDestroyerDefNoDestroyerWW2v2FalseSneakAttackFalse() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, false, false, false, false),
+        MustFightBattle.ReturnFire.ALL,
+        false);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has no destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker>defender>standard")
```

Comment:
nit, if the value of `and DEFENDING_SUBS_SNEAK_ATTACK is either`, you can probably just omit it from the display name. I think it would only be important if this test were parameterized and you actually test both the `DEFENDING_SUBS_SNEAK_ATTACK == true` and `DEFENDING_SUBS_SNEAK_ATTACK == false` cases.

Have you considered actually parameterizing this test to verify both of those cases? EG:

```
@ParameterizedTest
@ValueSource(booleans = { true, false })
void testMethodName(boolean defendingSubsSneakAttack) {
  // ....
}
```

If the test is parameterized, there is some auto-generation of test case names, check how they read in your IDE output and perhaps the display name can be shortened.